### ISSUE TITLE
Failure to upgrade database to DB v17 for MMEX 1.6.2

### ISSUE THREAD

**Issue by tactilis** [moneymanagerex/moneymanagerex#5433]
MMEX version:
 - [x] 1.6.2 Beta
 
Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

MMEX forum user @nollergelma got this error when trying to upgrade to 1.6.2 Beta 2.
 
![image](https://user-images.githubusercontent.com/4060092/208145430-0bea7d58-ceb6-48c7-a3ec-d5ff70224874.png)

See this thread [https://forum.moneymanagerex.org/viewtopic.php?p=23609#p23609](https://forum.moneymanagerex.org/viewtopic.php?p=23609#p23609).

@whalley, @n-stein are you able to assist please?

**Comment by n-stein** [comment]
Solution is in #5478. Use this patch file https://github.com/moneymanagerex/moneymanagerex/files/10087169/renato_db_fix.mmdbg.zip

**Comment by whalley** [comment]
Looks like the view name is slightly different... This may be better.
[views_db_fix.zip](https://github.com/moneymanagerex/moneymanagerex/files/10247625/views_db_fix.zip)

**Comment by n-stein** [comment]
Can we add these three "DROP VIEW" lines to the v17 upgrade script? I don't think this would cause any issues even though v17 was already rolled out, and it could prevent a lot of potential issues when 1.6.2 is released officially.

**Comment by tactilis** [comment]
Where did the views `alldata` and `alldatax` come from?

**Comment by n-stein** [comment]
Based on comments in #5378 it seems to be a relic of older versions of MMEX, though I do not know exactly when this was phased out. I believe its purpose was to provide a single datasource to facilitate easier creation of GRM reports and ODBC queries in Excel. I know some users have asked for it to be returned, such as in #3277.

**Comment by whalley** [comment]
https://github.com/moneymanagerex/database/commit/680a1f02283b7a02980af4971b9be65f55559d16

**Comment by renato-mmex** [comment]
@whalley where can I download this latest version
(this Version https://ci.appveyor.com/project/whalley/moneymanagerex/builds/45701544 does not work)

**Comment by renato-mmex** [comment]
Solved the problem with DROP VIEW alldata (with the DB Browser)

**Comment by whalley** [comment]
> @whalley where can I download this latest version (this Version https://ci.appveyor.com/project/whalley/moneymanagerex/builds/45701544 does not work)

Yes, the change was not yet incorporated in the MMES build. It is now.

**Comment by renato-mmex** [comment]
![Daumen](https://user-images.githubusercontent.com/76920138/208265389-55650fcf-8313-4201-b942-f52ab7db58ec.gif)

**Comment by renato-mmex** [comment]
Downloaded this version but still get the error:
<img width="411" alt="image" src="https://user-images.githubusercontent.com/76920138/208265615-4237e538-2a2a-4b3c-95ae-e09828dac7cc.png">

**Comment by tactilis** [comment]
Did you download the 'drop old views' build?

![image](https://user-images.githubusercontent.com/4060092/208267380-cb34a844-1307-45c4-8342-f7eab17b5611.png)

**Comment by renato-mmex** [comment]
Downloaded this version but still get the error:
https://ci.appveyor.com/project/whalley/moneymanagerex/builds/45712149/job/r529diae4byaha7m/artifacts

Merge pull request #5436 from whalley/master
fix(#5433): drop old views

3 hours ago by [Mark Whalley](https://github.com/whalley) (committed by [GitHub](https://github.com/web-flow))
[master](https://github.com/moneymanagerex/moneymanagerex/tree/master) [1a81cd26](https://github.com/moneymanagerex/moneymanagerex/commit/1a81cd262598f1c25bffb8a4ec1e4a4f5451a871)
2 hours ago in 20 min 12 sec
Environment: comp=141, wxShared=1; Platform: x64

**Comment by whalley** [comment]
Yes think that is my fault. I should have dropped the views first not at end of script. Let me try again tomorrow. 😀

**Comment by renato-mmex** [comment]
@whalley I had the same idea last night. This morning I looked at the file database_version_17.sql and got the same result.

**Comment by renato-mmex** [comment]
Gratulation, i have tested

<img width="345" alt="image" src="https://user-images.githubusercontent.com/76920138/208301493-b181f868-3d75-492d-a128-f0b006429f23.png">

![Daumen](https://user-images.githubusercontent.com/76920138/208301528-f2004dcd-b6e5-4881-af43-e44cba30206f.gif)
